xbmc/settings/AdvancedSettings.cpp

```diff
@@ -1199,6 +1199,12 @@ void CAdvancedSettings::ParseSettingsFile(const std::string &file)
       m_seekSteps.push_back(atoi((*it).c_str()));
   }
 
+  pElement = pRootElement->FirstChildElement("suspend");
+  if (pElement)
+  {
+    XMLUtils::GetBoolean(pElement, "waitnetonwakeup", m_waitForNetAfterWakeUp);
+  }
```

Comment:
`CAdvancedSettings::ParseSettingsFile()` is ridiculously long and should be refactored into local free functions.
That is too much work for a single feature implementation, but not adding more inline code would be a start.

Consider putting lines 1202 - 1206 into a free function
```
namespace {
	bool GetWaitNetOnWakeupFlag(TiXmlElement *rootElement)
	{
		if (!rootElement)
			throw std::invalid_argument("rootElement");

		auto firstChild = rootElement->FirstChildElement("suspend");
		if (!firstChild)
			return true; // Default behavior is to enable waitnetonwakeup

		bool waitForNetAfterWakeUp = true;
		XMLUtils::GetBoolean(firstChild, "waitnetonwakeup", waitForNetAfterWakeUp);
		return waitForNetAfterWakeUp;
	}
}
```
The line `1202: m_waitForNetAfterWakeUp = GetWaitNetOnWakeupFlag();` makes clear that the member is always well defined. Now, this is a lot more code than the original, but it is self-contained. `pRootElement` is checked for NULL, but this happens in line 444, i.e. 750+ lines above. Any (future changes) in between could invalidate it.

BTW.: IMHO, `XMLUtils` should provide function-style Getters, too, that don't require handing over a reference variable.

Yes, I'm new to this group and I do the code review like I do at work. Am I pushing too hard?